internal/userutils/userutils_test.go

```diff
@@ -0,0 +1,126 @@
+package userutils_test
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/userutils"
+	"github.com/ubuntu/authd/log"
+)
+
+var useSudo bool
+var gpasswdHelperPath string
+var groupFile string
+
+func TestLockAndUnlockGroupFile(t *testing.T) {
+	// This test requires either root privileges, unprivileged user namespaces, or being
+	// able to run sudo without user interaction.
+	if os.Geteuid() == 0 {
+		log.Info(context.Background(), "Running as EUID 0")
+	} else if canUseUnprivilegedUserNamespaces() {
+		log.Info(context.Background(), "Can use unprivileged user namespaces")
+	} else if canUseSudoNonInteractively() {
+		log.Info(context.Background(), "Can use sudo non-interactively")
```

Comment:
Use t.log here instead, we don't want to bother if -v  isn't used as the result should be enough.